selfhost/utility.jakt

```diff
@@ -74,6 +74,14 @@ fn join(anon strings: [String], separator: String) throws -> String {
     return output
 }
 
+fn map<T, U>(input: [T], mapper: fn(item: T) throws -> U) throws -> [U] {
```

Comment:
Please take this function as a reference so it'll play nice with trailing closures
with the trailing closure parameter support in #1474, the following will be possible if `map` takes a `&fn` instead of `fn`.
```
map(stuff) |item: i32| {
    return item + 1
}
```